src/main/java/org/terracotta/statistics/PassThroughStatistic.java

```diff
@@ -15,28 +15,29 @@
  */
 package org.terracotta.statistics;
 
+import org.terracotta.context.WeakIdentityHashMap;
+import org.terracotta.context.annotations.ContextAttribute;
+import org.terracotta.statistics.extended.StatisticType;
+
+import java.io.Serializable;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
-import java.util.concurrent.Callable;
 import java.util.concurrent.CopyOnWriteArrayList;
 
-import org.terracotta.context.WeakIdentityHashMap;
-import org.terracotta.context.annotations.ContextAttribute;
-
 @ContextAttribute(value="this")
-class PassThroughStatistic<T extends Number> implements ValueStatistic<T> {
+class PassThroughStatistic<T extends Serializable> implements ValueStatistic<T> {
 
-  private static final WeakIdentityHashMap<Object, Collection<PassThroughStatistic<?>>> BINDING = new WeakIdentityHashMap<>();
+  private static final WeakIdentityHashMap<Object, Collection<PassThroughStatistic<? extends Serializable>>> BINDING = new WeakIdentityHashMap<>();
   
-  private static void bindStatistic(PassThroughStatistic<?> stat, Object to) {
-    Collection<PassThroughStatistic<?>> collection = BINDING.get(to);
+  private static <T extends Serializable> void bindStatistic(PassThroughStatistic<T> stat, Object to) {
```

Comment:
Why do you need this type parameter?

Reply:
I'll move back - no reason - I thought after if we add a new param typed with the same T... But anyway this will be a signature change, so...